cpp/src/parquet/arrow/reader.h

```diff
@@ -208,6 +231,16 @@ class RowGroupReader {
   virtual ::arrow::Status ReadTable(const std::vector<int>& column_indices,
                                     std::shared_ptr<::arrow::Table>* out) = 0;
   virtual ::arrow::Status ReadTable(std::shared_ptr<::arrow::Table>* out) = 0;
+
+  ::arrow::Status MakeIterator(
+      std::unique_ptr<::arrow::Iterator<std::shared_ptr<::arrow::RecordBatch>>>* batches);
+
+  ::arrow::Status MakeIterator(
+      const std::vector<int>& column_indices,
+      std::unique_ptr<::arrow::Iterator<std::shared_ptr<::arrow::RecordBatch>>>* batches);
```

Comment:
I think this needs to be reconciled with the GetRecordBatchReader function above

Reply:
do you mean rename it to `GetRecordBatchIterator`?